File: Code/java/SLP/red_add.java
import java.util.Arrays;
class red_add {
 public static  int [] arr;
 public static int micro(int ctr) {
    int res = 0;
    for(int i = 0; i < arr.length; i++) {
        res = res + arr[i] * (2);
    }
    return res;
 }
 public static void main(String [] args) {
    int res = 0;
    arr = new int[1024];
    Arrays.fill(arr, 10);
    for(int i = 0; i < 100000; i++) {
       res += micro(i & 1023);     
    }
    System.out.println(res);
 }

}

File: Code/scala/red_add.scala

import java.util.Arrays

object red_add {
   def micro(arr : Array[Int]) : Int = {
      var i : Int = 0
      var res : Int = 0
      var elem : Int = 0
      for (elem <- arr) {
         res += arr(i) 
      }
      return res
   }
   
   def main(args : Array[String]) {
      var res : Int = 0
      var arr : Array[Int] = new Array[Int](1024)
      Arrays.fill(arr, 10)
      var i : Int = 0
      for (i <- 0 to 10000) {
         res += micro(arr)
      } 
      println("[res]  " + res)
   }
}
